decidim-elections/app/forms/decidim/elections/admin/setup_form.rb

```diff
@@ -13,17 +13,15 @@ class SetupForm < Decidim::Form
 
         def map_model(model)
           @election = model
-          @trustees = Decidim::Elections::Trustee.includes([:user]).joins(:trustees_participatory_spaces).merge(Decidim::Elections::TrusteesParticipatorySpace
-                                                                                                         .where(participatory_space: election.component.participatory_space,
-                                                                                                                considered: true)).to_a.sample(number_of_trustees).sort_by(&:id)
+          @trustees = Decidim::Elections::Trustees::ByParticipatorySpace.new(election.component.participatory_space).to_a.sample(number_of_trustees).sort_by(&:id)
+
           self.trustee_ids = @trustees.pluck(:id)
         end
 
         def trustees
-          @trustees ||= Decidim::Elections::Trustee.includes([:user]).joins(:trustees_participatory_spaces).merge(Decidim::Elections::TrusteesParticipatorySpace
-                                                                                                           .where(decidim_elections_trustee_id: trustee_ids,
-                                                                                                                  participatory_space: election.component.participatory_space,
-                                                                                                                  considered: true)).to_a.sort_by(&:id)
+          @trustees = Decidim::Elections::Trustees::ByParticipatorySpace.new(election.component.participatory_space)
+          @trustees = Decidim::Elections::Trustees::ByParticipatorySpaceTrusteeIds.new(trustee_ids).to_a.sort_by(&:id)
```

Comment:
Sorry, I don't see why you are assigning the `@trustees` variable twice. I'd say that the whole method would work the same if we only write the following line, but maybe I'm missing something. 🤔 
```ruby
@trustees ||= Decidim::Elections::Trustees::ByParticipatorySpaceTrusteeIds.new(trustee_ids).to_a.sort_by(&:id)
```

Reply:
Because these are two different queries. The first is the one for the `participatory_space` and the other for the `trustee_ids`.

Reply:
yes, but you are storing both results on the same `@trustees` variable, and I'm not able to see how are you using the first result on the second query. Maybe you should do a `self.trustee_ids = @trustees.pluck(:id)` before running the second query?